test/runtests.jl

```diff
@@ -80,25 +87,142 @@ facts("SDDP algorithm") do
     # set bounds on control:
     u_bounds = [(0., 7.), (0., Inf)]
 
-    # Instantiate a SDDP linear model:
-    model = StochDynamicProgramming.LinearDynamicLinearCostSPmodel(n_stages,
-                                                2, 1, 1,
-                                                x_bounds, u_bounds, x0,
-                                                cost,
-                                                dynamic, laws)
+    # Instantiate parameters of SDDP:
+    params = StochDynamicProgramming.SDDPparameters(solver, n_scenarios,
+                                                    epsilon, max_iterations)
+
+
+    context("Linear cost") do
+        # Instantiate a SDDP linear model:
+        model = StochDynamicProgramming.LinearDynamicLinearCostSPmodel(n_stages,
+                                                    u_bounds, x0,
+                                                    cost,
+                                                    dynamic, laws)
+        set_state_bounds(model, x_bounds)
+
+
+        # Compute bellman functions with SDDP:
+        V, pbs = solve_SDDP(model, params, false)
+        @fact typeof(V) --> Vector{StochDynamicProgramming.PolyhedralFunction}
+        @fact typeof(pbs) --> Vector{JuMP.Model}
+
+        # Test if the first subgradient has the same dimension as state:
+        @fact length(V[1].lambdas[1, :]) --> model.dimStates
+        @fact V[1].numCuts --> n_stages*n_scenarios + 1
+        @fact length(V[1].lambdas[:, 1]) --> n_stages*n_scenarios + 1
+
+        # Test upper bounds estimation with Monte-Carlo:
+        n_simulations = 100
+        upb = StochDynamicProgramming.estimate_upper_bound(model, params, V, pbs,
+                                                           n_simulations)[1]
+        @fact typeof(upb) --> Float64
+
+        # Test a simulation upon given scenarios:
+        aleas = simulate_scenarios(model.noises,
+                              (model.stageNumber,
+                               params.forwardPassNumber,
```

Comment:
Use n_simulations instead, else you are checking a monte carlo result with 10 simulation.